classes/various/Matrix.sc

```diff
@@ -181,6 +181,30 @@ Matrix[slot] : Array {
 		this.putCol(posA, this.getCol(posB) );
 		this.putCol(posB, colA);
 	}
+	/*
+	Michael McCrea, 2019.
+	*/
+	// this is a destructive operation:
+	// force values to zero that are within threshold distance (positive or negative)
+	zeroWithin { |within = (-180.dbamp)|
+		this.rowsDo({ |rArray, ri|
+			rArray.do({ |item, ci|
+				this.put(
+					ri,
+					ci,
+					if(item.abs <= within, {
+						if(item.isInteger, {  // there could be more cases...
```

Comment:
if there were more cases it would be better to make method like `Integer::zeroFor` etc.

Reply:
I'm not sure why I'd added that comment originally, @joslloand,  do you know what cases we may have been considering here?

Reply:
@mtmccrea, I expect we were considering `Complex` so as to do more fancy things with HOA-NFE, for instance:

- design decoders against measured radial responses
- design HRTF decoders
- estimate loudspeaker source / microphone receiver distance

We haven't gotten around to these problems, as they really are _complex_! I'd need to revisit what we've got in Python, to see what functionality would need to be ported, and how best to express things with `Matrix`.

That's my thinking, anyway.

Reply:
maybe, and an intermediate thing, there could be a special `Complex`  subclass.